src/Pages/SignIn/SignIn.jsx

```diff
@@ -0,0 +1,120 @@
+import { Link, useNavigate } from "react-router-dom";
+import { useState, useEffect } from "react";
+import "./SignIn.css";
+import { useAuth } from "../../Context/loginContext";
+import googleLogo from "../../assets/images/GoogleLogo.png";
+import { GoogleAuthProvider, signInWithEmailAndPassword, signInWithPopup } from "firebase/auth";
+import { firebaseAuth } from "../../firebase.config";
+export function SignIn() {
+  const { login, setLogin } = useAuth();
+  const navigate = useNavigate();
+  const googleProvider = new GoogleAuthProvider();
+  const [loginData, setLoginData] = useState({
+    email: "",
+    password: "",
+  });
+  const onChangeHandler = (e) => {
+    setLoginData((loginData) => ({
+      ...loginData,
+      [e.target.name]: e.target.value,
+    }));
+  };
+  const signInHandler = async (e) => {
+    e.preventDefault();
+    try {
+      let response = await signInWithEmailAndPassword(
+        firebaseAuth,
+        loginData.email,
+        loginData.password
+      );
+      localStorage.setItem("login", response.user.accessToken);
+      localStorage.setItem("user data", response.user.providerData[0]);
+      setLogin(true);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  async function googleSignInHandler() {
+    try {
+      const response = await signInWithPopup(firebaseAuth, googleProvider);
+
+      localStorage.setItem("login", response.user.accessToken);
+      localStorage.setItem("user data", response.user.providerData[0]);
+      setLogin(true);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  useEffect(() => {
+    login && navigate("/");
+  }, []);
+
+  return (
+    <div className="show-sign-in">
+      <div className="form-container">
+        <div className="form-subcontainer">
+          <h3 className="form-title">Sign-In</h3>
+
+          <form action="" className="signIn-form" onSubmit={signInHandler}>
+            <label
+              htmlFor="email-input"
+              className="form-label"
+              id="email-label"
+            >
+              Email
+            </label>
+            <input
```

Comment:
Suggestion: You could abstract form login or signup input components in a separate component to make the code more modular